app/src/org/commcare/activities/DispatchActivity.java

```diff
@@ -114,8 +115,8 @@ protected void onSaveInstanceState(Bundle outState) {
 
     private void checkForChangedCCZ() {
         alreadyCheckedForAppFilesChange = true;
-        Intent i = new Intent(getApplicationContext(), UpdateActivity.class);
-        startActivity(i);
+        Intent i = new Intent(this, UpdateActivity.class);
+        startActivityForResult(i, UPDATE_CONSUMER_APP);
```

Comment:
Ah yea you're right, this was one of the things I tried to resolve the bug, but I'll get rid of it